File: MonoEngineV2/Vector3f.cpp
#include "pch.h"
#include "Vector3f.h"

Vector3f::Vector3f(_In_ float x, _In_ float y, _In_ float z)
	: x(x), y(y), z(z)
{ }

Vector3f::Vector3f(_In_ float x, _In_ float y)
	: x(x), y(y), z(0.0f)
{ }

Vector3f::Vector3f()
	: Vector3f(0.0f, 0.0f, 0.0f)
{ }

File: MonoEngineV2Lib/Vector3f.cs
using System.Runtime.InteropServices;

namespace MonoEngineV2Lib
{
    [StructLayout(LayoutKind.Sequential)]
    public struct Vector3f
    {
        public float x;
        public float y;
        public float z;

        public Vector3f(float _x, float _y, float _z)
        {
            x = _x;
            y = _y;
            z = _z;
        }

        public Vector3f(float _x, float _y)
            : this(_x, _y, 0.0f)
        { }

        public override string ToString()
        {
            return $"{{x={x}, y={y}, z={z}}}";
        }
    }
}